modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -50,6 +54,15 @@ func (a TransferAuthorization) Accept(ctx context.Context, msg proto.Message) (a
 			return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidAddress, "not allowed receiver address for transfer")
 		}
 
+		isAllowedPacket, err := isAllowedPacketDataKeys(sdk.UnwrapSDKContext(ctx), msgTransfer.Memo, allocation.AllowPacketData)
+		if err != nil {
+			return authz.AcceptResponse{}, errorsmod.Wrapf(ErrInvalidMemo, "error: %v", err)
```

Comment:
typically I recommend wrapping the base error, especially when we produce the base error, this avoids switching up the order of the trace.

Reply:
I think the errors returned from `validateMemo` look well formulated enough now to return directly here? wdyt?

```suggestion
			return authz.AcceptResponse{}, err
```